---
title: "Differential Gene Expression Analysis using DESeq2"
author: "Yogesh Purushotham"
date: "2024-02-22"
output: html_document
---

# Introduction

This analysis aims to investigate differential gene expression using the DESeq2 package, focusing on the response to dexamethasone treatment.

## Set Working Directory

```{r setup, echo=FALSE}
# setwd("~/Desktop/work/DESeq2/data")

```

## Loading libraries required

```{r cars}
# load libraries
library(DESeq2)
library(tidyverse)
library(airway)
```


## Script to get data from airway package

```{r}

data(airway)
airway

sample_info <- as.data.frame(colData(airway))
sample_info <- sample_info[,c(2,3)]
sample_info$dex <- gsub('trt', 'treated', sample_info$dex)
sample_info$dex <- gsub('untrt', 'untreated', sample_info$dex)
names(sample_info) <- c('cellLine', 'dexamethasone')
write.table(sample_info, file = "sample_info.csv", sep = ',', col.names = T, row.names = T, quote = F)

countsData <- assay(airway)
write.table(countsData, file = "counts_data.csv", sep = ',', col.names = T, row.names = T, quote = F)

```

## Preparing count data 

```{r}
# reading in counts data
counts_data <- read.csv('counts_data.csv')
head(counts_data)

# reading in sample info
colData <- read.csv('sample_info.csv')

# making sure the row names in colData matches to column names in counts_data
all(colnames(counts_data) %in% rownames(colData))

# To check if they are in the same order
all(colnames(counts_data) == rownames(colData))

```

## Constructing a DESeqDataSet object:

```{r}

dds <- DESeqDataSetFromMatrix(countData = counts_data,
                       colData = colData,
                       design = ~ dexamethasone)
dds

# pre-filtering/removing rows with low gene counts
# keeping rows that have at least 10 reads total
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
dds

# setting the factor level
dds$dexamethasone <- relevel(dds$dexamethasone, ref = "untreated")

```


## Running DESeq 

```{r}

dds <- DESeq(dds)
res <- results(dds)

res

```

## Exploring Results

```{r}
summary(res)

res0.01 <- results(dds, alpha = 0.01)
summary(res0.01)

# contrasts
resultsNames(dds)

# e.g.: treated_4hrs, treated_8hrs, untreated
results(dds, contrast = c("dexamethasone", "treated_4hrs", "untreated"))

# MA plot
plotMA(res)

```
